src/Microsoft.NetCore.Analyzers/Core/Runtime/DoNotLockOnObjectsWithWeakIdentity.cs

```diff
@@ -51,10 +52,18 @@ public override void Initialize(AnalysisContext analysisContext)
                 compilationStartContext.RegisterOperationAction(context =>
                 {
                     var lockStatement = (ILockOperation)context.Operation;
-                    ITypeSymbol type = lockStatement.LockedValue?.Type;
-                    if (type != null && TypeHasWeakIdentity(type, compilation))
+
+                    if (lockStatement.LockedValue.Kind == OperationKind.InstanceReference)
+                    {
+                        context.ReportDiagnostic(lockStatement.LockedValue.Syntax.CreateDiagnostic(Rule, lockStatement.LockedValue.Syntax.ToString()));
```

Comment:
There was nothing to help me know whether the argument specified is used for something in particular so I have decided to pass the `this` or `Me` instead of the type in this case. This can be changed to also pass the type if needed.

Reply:
Seems fine to me.